server/index-node/src/resolver.rs

```diff
@@ -176,14 +214,60 @@ impl TryFromValue for IndexingStatusWithoutNode {
 
 impl From<IndexingStatus> for q::Value {
     fn from(status: IndexingStatus) -> Self {
+        let IndexingStatus {
+            subgraph,
+            chains,
+            fatal_error,
+            health,
+            node,
+            non_fatal_errors,
+            synced,
+        } = status;
+
+        fn subgraph_error_to_value(subgraph_error: SubgraphError) -> q::Value {
+            let SubgraphError {
+                subgraph_id,
+                message,
+                block_ptr,
+                handler,
+            } = subgraph_error;
+            let block_value = {
+                let mut block_entity = Entity::new();
+                block_entity.set("__typename".to_string(), "Block".to_string());
+                block_entity.set("number".to_string(), block_ptr.map(|x| x.number));
+                block_entity.set("hash".to_string(), block_ptr.map(|x| x.hash));
+                q::Value::from(block_entity)
+            };
+
+            let mut error_object = {
+                let mut error_entity = Entity::new();
+                error_entity.set("__typename".to_string(), "SubgraphError".to_string());
+                error_entity.set("subgraphId".to_string(), subgraph_id.to_string());
+                error_entity.set("message".to_string(), message);
+                error_entity.set("handler".to_string(), handler);
+                BTreeMap::from(error_entity)
+            };
+
+            error_object.insert("block".to_string(), block_value.into());
+
+            q::Value::Object(error_object)
```

Comment:
Since we have `entity!` now, that could be a cool way to clean this file up a little:
```rust
let error_object = entity! {
    ...
};
```
may already do the job, may not even have to convert it to a `q::Value::Object` explicitly, if `object!` does it for us (I'm not sure).

Reply:
This is way better!